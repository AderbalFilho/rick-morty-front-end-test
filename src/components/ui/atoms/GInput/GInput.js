import React from 'react';
import classNames from 'classnames/bind';
import PropTypes from 'prop-types';

import './GInput.scss';

function GInput({ change, classes, ...rest }) {
  const className = classNames('g-input', classes);

  return <input className={className} onChange={change} {...rest} />;
}

GInput.defaultProps = {
  change: () => {},
};

GInput.propTypes = {
  change: PropTypes.func,
  classes: PropTypes.string,
};

export default GInput;
