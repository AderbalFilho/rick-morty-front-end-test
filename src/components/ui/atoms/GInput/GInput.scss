@import 'styles/mixins';

.g-input {
  @include base-text;

  background-color: transparent;
  border: 2px solid var(--color-white);
  border-radius: var(--radius-medium);
  font-weight: bold;
  max-width: 196px;
  padding: 10px 14px;

  &:focus {
    outline: none;
  }

  &::placeholder {
    color: var(--color-dark-grey);
  }
}
