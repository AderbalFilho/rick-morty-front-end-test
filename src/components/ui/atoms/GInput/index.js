import GInput from './GInput';

export default GInput;
