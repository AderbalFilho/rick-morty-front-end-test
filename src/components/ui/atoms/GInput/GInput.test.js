import React from 'react';
import TestRenderer from 'react-test-renderer';
import { render, fireEvent } from '@testing-library/react';

import GInput from './GInput';

describe('GInput', () => {
  it('matches snapshot', () => {
    const testRenderer = TestRenderer.create(<GInput />);
    const jsonComponent = testRenderer.toJSON();

    expect(jsonComponent).toMatchSnapshot();
  });

  it('fire event when input text changed', () => {
    const onChange = jest.fn();
    const { getByPlaceholderText } = render(
      <GInput type="text" placeholder="Search characters" change={onChange} />
    );

    fireEvent.change(getByPlaceholderText(/Search characters/i), {
      target: { value: 'inGaia' },
    });

    expect(onChange).toHaveBeenCalled();
  });

  it('renders classes', () => {
    const { container } = render(<GInput type="text" classes="input-search" />);
    const card = container.querySelector('.g-input');

    expect(card.classList).toContain('input-search');
  });
});
