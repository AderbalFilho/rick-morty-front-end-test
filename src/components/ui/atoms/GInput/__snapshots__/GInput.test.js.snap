// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GInput matches snapshot 1`] = `
<input
  className="g-input"
  onChange={[Function]}
/>
`;
